lib/modules/modal.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+const T = React.PropTypes;
+
+const STYLE_CONTAINER = {
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  right: 0,
+  bottom: 0,
+  zIndex: 1000,
+  backgroundColor: 'rgba(0,0,0,0.4)',
+};
+
+// ******* ACTIONS
+export const TOGGLE_MODAL = 'r/widgets__TOGGLE_MODAL';
+
+export const toggleModal = id => ({
+  type: TOGGLE_MODAL,
+  payload: { id },
+});
+
+// ******* RAW COMPONENTS
+export class _Modal extends React.Component {
+  static propTypes = {
+    className: T.string,
+    show: T.bool,
+    onToggleModal: T.func,
+  };
+
+  static defaultProps = {
+    className: '',
+    show: false,
+    onToggleModal: () => {},
+  };
+
+  render() {
+    const { className, show, children, onToggleModal } = this.props;
+
+    if (show) {
+      return (
+        <div style={ STYLE_CONTAINER } onClick={ () => onToggleModal(null) }>
+          <div className={ className } onClick={ e => e.stopPropagation() }>
+            { children }
+          </div>
+        </div>
+      );
+    } else {
+      return null;
+    }
+  }
+}
+
+export class _ModalTarget extends React.Component {
+  static propTypes = {
+    onToggleModal: T.func,
```

Comment:
It's probably good to add `children` to propTypes here because you've modified its signature